examples/games/desktoy.rs

```diff
@@ -0,0 +1,320 @@
+        primary_window.cursor.hit_test = true;
+        return;
+    }
+
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    let bevy_logo_transform = q_bevy_logo.single();
+    primary_window.cursor.hit_test =
+        (bevy_logo_transform.translation.truncate() - cursor_world_pos).length() < BEVY_LOGO_RADIUS;
+}
+
+/// Start the drag operation and record the offset we started dragging from
+fn start_drag(
+    mut commands: Commands,
+    cursor_world_pos: Res<CursorWorldPos>,
+    q_bevy_logo: Query<&Transform, With<BevyLogo>>,
+) {
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    let bevy_logo_transform = q_bevy_logo.single();
+    let drag_offset = bevy_logo_transform.translation.truncate() - cursor_world_pos;
+    if drag_offset.length() < BEVY_LOGO_RADIUS {
+        commands.insert_resource(DragOffset(drag_offset));
+    }
+}
+
+/// Stop the current drag operation
+fn end_drag(mut commands: Commands) {
+    commands.remove_resource::<DragOffset>();
+}
+
+/// Drag the Bevy logo
+fn drag(
+    drag_offset: Res<DragOffset>,
+    cursor_world_pos: Res<CursorWorldPos>,
+    time: Res<Time>,
+    mut q_bevy_logo: Query<&mut Transform, With<BevyLogo>>,
+    mut q_pupils: Query<&mut Pupil>,
+) {
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    let mut bevy_transform = q_bevy_logo.single_mut();
+    let new_translation = cursor_world_pos + drag_offset.0;
+    let drag_velocity =
+        (new_translation - bevy_transform.translation.truncate()) / time.delta_seconds();
+    bevy_transform.translation = new_translation.extend(bevy_transform.translation.z);
+
+    for mut pupil in &mut q_pupils {
+        pupil.velocity -= drag_velocity;
+    }
+}
+
+/// Quit when the user right clicks the Bevy logo
+fn quit(
+    cursor_world_pos: Res<CursorWorldPos>,
+    mut app_exit: EventWriter<AppExit>,
+    q_bevy_logo: Query<&Transform, With<BevyLogo>>,
+) {
+    let Some(cursor_world_pos) = cursor_world_pos.0 else {
+        return;
+    };
+
+    let bevy_logo_transform = q_bevy_logo.single();
+    if (bevy_logo_transform.translation.truncate() - cursor_world_pos).length() < BEVY_LOGO_RADIUS {
+        app_exit.send(AppExit);
+    }
+}
+
+/// Enable transparency for the window and make it on top
+fn enable_transparency(
+    mut commands: Commands,
+    q_instructions_text: Query<Entity, With<InstructionsText>>,
+    mut q_primary_window: Query<&mut Window, With<PrimaryWindow>>,
+) {
+    let _ = q_instructions_text
+        .get_single()
+        .map(|entity| commands.entity(entity).despawn_recursive());
+    let mut window = q_primary_window.single_mut();
+    window.decorations = false;
+    window.window_level = WindowLevel::AlwaysOnTop;
+    commands.insert_resource(ClearColor(Color::NONE));
+}
+
+/// Bounce the pupils around
+fn update_pupils(time: Res<Time>, mut q_pupils: Query<(&mut Pupil, &mut Transform)>) {
```

Comment:
Would be much clearer as `bounce_pupils`.

Reply:
I changed it to `move_pupils` since it handles both movement and bouncing.